### DISCUSSION TITLE
[Monitoring] Request/Limit Microservices

### DISCUSSION THREAD

**Discussion by SalvadorCali** [mia-platform/community#206] [Q&A] [answered]
Hi, everyone!
We need to monitor our services CPU and memory request and limit to check if they are correctly set. 
Using Prometheus we risk to lose any peaks during the collection of the data.

Do you have any ideas?

**Comment by fredmaggiowski** [comment, marked as answer]
Hi, what do you mean by "monitor request and limit"? They are static values, do you mean monitor usage to better tune request and limits?

There are other tools other than prometheus that may help you but apart from that I don't think you really need to care about peaks, you need to be aware of them and eventually make your infrastructure flexible enough to be able to scale if a peak becomes too long; but generally you want to tune your request and limit based on the normal usage (adding replicas if there is a sudden increase in traffic). 

If the sudden increase in traffic occurs more often you might want to rethink what your normal usage is and eventually increase resource requirements/scaling factor or try to optimize your application

**Reply by SalvadorCali** [reply]
Hi!
> They are static values, do you mean monitor usage to better tune request and limits?

Yes, I expressed the problem incorrectly.

Our goal is to have a report/alarm that inform us if one of our services is using more/less resources then those expected. 
Below two situations that led us to this need:

- one of our services started to receive a lot of requests that wasn't expected during the tuning of request/limits
- during full refresh our services use an amount of resources much greater than during the basic flow

**Reply by fredmaggiowski** [reply]
> one of our services started to receive a lot of requests that wasn't expected during the tuning of request/limits

Aren't throttle alarm enough? You could add custom metrics on invoked endpoints in your microservices to have counters and define a rps based alarm but this seems a bit overkill to me 🤔 

> during full refresh our services use an amount of resources much greater than during the basic flow

if these operations are scheduled you might decide to manually scale the services beforehand?
